src/server/auth/server/api_server.go

```diff
@@ -543,7 +543,10 @@ func (a *apiServer) AuthorizeInTransaction(
 
 	// Get the role bindings for the resource to check
 	var roleBinding auth.RoleBinding
-	if err := a.roleBindings.ReadWrite(txnCtx.Stm).Get(resourceKey(req.Resource), &roleBinding); err != nil && !col.IsErrNotFound(err) {
```

Comment:
If the role binding doesn't exist in Authorize, throw ErrNoRoleBinding rather than trying to evaluate the cluster binding alone. Every resource should always have a corresponding role binding.